php-checks/src/test/resources/checks/NestedSwitchCheck.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+switch($a) { // Compliant
+  case "foo":
+    switch($b) { // Noncompliant
```

Comment:
You may add an "assertion" about the precise location of the issue. See https://github.com/SonarSource/sonar-analyzer-commons/tree/master/test-commons
However, checking the precise location of the issue should not be done for all the test cases of the file, only when it makes sense.